File: sliding_window/209. Minimum Size Subarray Sum.cpp
#include <bits/stdc++.h>

class Solution {
public:
    int minSubArrayLen(int target, std::vector<int>& nums) {
        int window_sum = 0,
            window_length = INT_MAX,
            window_start = 0;

        for (int window_end = 0 ; window_end < nums.size(); window_end++) {
            window_sum += nums[window_end];

            while (window_sum >= target) {
                window_length = std::min(window_length, window_end - window_start + 1);
                window_sum -= nums[window_start];
                window_start++;
            }
        }

        if (window_length == INT_MAX) return 0;

        return window_length;
    }
};

File: sliding_window/209. Minimum Size Subarray Sum.py
from typing import List

class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        min_length = float('inf') 
        window_start = window_sum = 0

        for window_end in range(len(nums)):
            window_sum += nums[window_end]

            while (window_sum >= target):
                min_length = min(min_length, window_end - window_start + 1)
                window_sum -= nums[window_start]
                window_start += 1

        return 0 if min_length == float('inf') else min_length

sol = Solution()
print(sol.minSubArrayLen(7, [2,3,1,2,4,3]))
